packages/components/src/shared/src/focusableTreeWalker.ts

```diff
@@ -16,14 +16,62 @@ const FocusableElement = [
     "[contenteditable]"
 ];
 
-export const FocusableElementSelector = [...FocusableElement, "[tabindex]"].join(",");
+export const FocusableElementSelector = FocusableElements.join(":not([hidden]),") + ",[tabindex]:not([disabled]):not([hidden])";
+FocusableElements.push("[tabindex]:not([tabindex=\"-1\"]):not([disabled])");
```

Comment:
I would change the synthax here so its more like before, instead of the spectrum synthax